File: leet-code/getMinimumDifference.go
package main

import "math"

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func getMinimumDifference(root *TreeNode) int {
	ans := math.MaxInt32
	pre := -math.MaxInt32

	var dfs func(root *TreeNode)
	dfs = func(root *TreeNode) {
		if root == nil {
			return
		}
		dfs(root.Left)
		ans = min(root.Val-pre, ans)
		pre = root.Val
		dfs(root.Right)
	}
	dfs(root)
	return ans
}

func min(a int, b int) int {

	if a < b {
		return a
	} else {
		return b
	}
}

File: leet-code/getMinimumDifference.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var getMinimumDifference = function (root) {
  let pre = -Number.MAX_SAFE_INTEGER
  let ans = Number.MAX_SAFE_INTEGER

  let dfs = function (root) {
    if (root == null) {
      return null
    }
    dfs(root.left)
    ans = Math.min(root.val - pre, ans)
    pre = root.val
    dfs(root.right)
  }
  dfs(root)
  return ans
}
